web/client/epics/__tests__/annotations-test.js

```diff
@@ -284,10 +284,18 @@ describe('annotations Epics', () => {
     it('set style', (done) => {
         store.subscribe(() => {
             const actions = store.getActions();
+            const state = {
+                annotations: {
+                    config: {
+                        geodesic: true
+                    }
+                }
+            };
             if (actions.length >= 2) {
                 expect(actions[0].type).toBe(SET_STYLE);
                 expect(actions[1].type).toBe(CHANGE_DRAWING_STATUS);
                 expect(actions[1].status).toBe("updateStyle");
+                expect(state.annotations.config.geodesic).toBe(true);
```

Comment:
this is wrong and it does not makes sense to test a local object 

I suggest to check how others tests are done with "testEpic" util function when they pass a mocked state
like ` it('MAP_CONFIG_LOADED with missing annotations layer', (done) => {`

Also why are you changing this test instead of editing the one we have discussed 
`it('clicked on map selecting a feature LineString ', (done) => {`

if you need clarification on how to write tests for the epics, please let me know

Reply:
this could be a solution for the test

```
it('clicked on map selecting a Circle ', (done) => {
        store = mockStore( {
            ...defaultState,
            annotations: {
                ...defaultState.annotations,
                config: {
                    geodesic: true
                },
                editing: {
                    ...defaultState.annotations.editing,
                    features: [{
                        type: "Feature",
                        geometry: {
                            type: "Polygon",
                            coordinates: [[[1, 2], [1, 3], [1, undefined], [1, 5], [1, 2]]]
                        },
                        properties: {
                            id: "is a circle",
                            isCircle: true
                        }
                    }
                    ]

                }
            }
        } );

        store.subscribe(() => {
            const actions = store.getActions();
            if (actions.length >= 3) {
                expect(actions[1].type).toBe(CHANGE_DRAWING_STATUS);
                expect(actions[2].type).toBe(CHANGE_DRAWING_STATUS);
                expect(actions[2].options.geodesic).toBe(true);
                done();
            }
        });
        const action = selectFeatures([ft]);
        store.dispatch(action);
    });
```

Reply:
I am  going through the testEpic and  I am understanding  how they done.